pkg/osd_cluster_upgrader/upgrader_test.go

```diff
@@ -190,6 +197,32 @@ var _ = Describe("ClusterUpgrader", func() {
 				Expect(result).To(BeFalse())
 			})
 		})
+
+		Context("When the cluster is not on the same channel as the UpgradeConfig", func() {
+			It("Updates the cluster's update channel", func() {
+				clusterVersionList := &configv1.ClusterVersionList{
+					Items: []configv1.ClusterVersion{
+						{
+							Spec: configv1.ClusterVersionSpec{
+								Channel: upgradeConfig.Spec.Desired.Channel + "not-the-same",
+								DesiredUpdate: nil,
+							},
+						},
+					},
+				}
+				expectUpgradeCommenced(mockKubeClient, clusterVersionList, nil)
+				util.ExpectGetClusterVersion(mockKubeClient, clusterVersionList, nil)
+				gomock.InOrder(
+					mockKubeClient.EXPECT().Update(gomock.Any(), gomock.Any()),
+					mockKubeClient.EXPECT().List(gomock.Any(), gomock.Any()).SetArg(1, *clusterVersionList),
+				)
+				result, err := CommenceUpgrade(mockKubeClient, config, mockScalerClient, mockMetricsClient, mockMaintClient, upgradeConfig, logger)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(result).To(BeFalse())
+
```

Comment:
`Expect(cv.Spec.Channel == upgradeConfig.Spec.Desired.Channel)`?

Reply:
Fixed in 6539f2c